Tasks/NpmAuthenticate/npmauth.ts

```diff
@@ -0,0 +1,136 @@
+import * as path from 'path';
+import * as tl from 'vsts-task-lib/task';
+import * as URL from 'url';
+import * as fs from 'fs';
+import * as constants_1 from './constants';
+import * as npmregistry_1 from 'npm-common/npmregistry'
+import * as util from 'npm-common/util';
+
+async function main(): Promise<void> {
+    tl.setResourcePath(path.join(__dirname, 'task.json'));
+    let saveNpmrcPath;
+    let npmrc = tl.getInput(constants_1.NpmAuthenticateTaskInput.WorkingDir);
+    let workingDir = path.dirname(npmrc);
+    if (!(npmrc.endsWith('.npmrc') && tl.exist(npmrc))) {
+        throw new Error(tl.loc('NpmrcNotNpmrc', npmrc));
+    }
+    else {
+        console.log(tl.loc("AuthenticatingThisNpmrc", npmrc));
+    }
+
+    if (process.env.SAVE_NPMRC_PATH) {
```

Comment:
Task variables are specific to the individual call of the task. If my build definition includes six npm Authenticate calls, each call will have its own task variables. They all share environment variables.